control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -212,7 +212,7 @@ func (r *GatewayController) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		r.gatewayCache.RemoveSubscription(nonNormalizedConsulKey)
 		// make sure we have deregister all services even if they haven't
 		// hit cache yet
-		if err := r.deregisterAllServices(ctx, consulKey); err != nil {
+		if err := r.deregisterAllServices(ctx, nonNormalizedConsulKey); err != nil {
```

Comment:
We use the `nonNormalizedConsulKey` everywhere in this section of code because it doesn't convert the `""` namespace to `"default"` and result in us setting a namespace on the query. Between this and the additional guard above, we're doubly careful that we aren't setting the namespace on a query unless
1. namespaces are enabled
2. the key has a namespace present